openmdao/core/driver.py

```diff
@@ -1744,3 +1780,62 @@ def record_iteration(requester, prob, case_name):
             data['rel'] = norm / norm0
 
     rec_mgr.record_iteration(requester, data, requester._get_recorder_metadata(case_name))
+
+
+def pass_by_meta(metadict_items, key, chk_none=False):
+    """
+    Yield only items with meta[key] of True or not None, depending on chk_none.
+
+    Parameters
+    ----------
+    metadict_items : iter of (name, meta)
+        Iterable of (name, meta) tuples.
+    key : str
+        Metadata key.
+    chk_none : bool
+        If True, yield items where meta[key] is not None. If False, yield items where meta[key] is
+        True.
+
+    Yields
+    ------
+    tuple
+        Tuple of the form (name, meta) for each item in metadict_items where meta[key] is True
+        or not None, depending on chk_none.
+    """
+    if chk_none:
+        for tup in metadict_items:
+            if tup[1][key] is not None:
+                yield tup
+    else:
+        for tup in metadict_items:
+            if tup[1][key]:
+                yield tup
+
+
+def filter_by_meta(metadict_items, key, chk_none=False):
+    """
+    Yield only items where meta[key] is not True.
```

Comment:
These functions are practically the same.  Seems like they could be generalized into one function, maybe with includes and excludes.  Just some thoughts.

Reply:
done